service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/SetCreateResponseStep.java

```diff
@@ -20,11 +23,38 @@ public StepResult doStep(FlightContext flightContext)
       throws InterruptedException, RetryException {
 
     // Return the input resource as the response; it is filled in properly on the request,
-    // so has all the right data. And if we have made it this far, we have made the request
-    // contents true.
+    // so has all the right data except the private resource state, if relevant. And if we have made
+    // it this far, we have made the request contents true.
     FlightMap workingMap = flightContext.getWorkingMap();
-    workingMap.put(JobMapKeys.RESPONSE.getKeyName(), resource);
-
+    ControlledResource responseResource;
+    if (resource.getAccessScope() == AccessScopeType.ACCESS_SCOPE_PRIVATE) {
+      switch (resource.getResourceType()) {
+        case GCS_BUCKET:
+          responseResource =
+              resource.castToGcsBucketResource().toBuilder()
+                  .privateResourceState(PrivateResourceState.ACTIVE)
```

Comment:
That's one of the annoying parts of our immutable resource object set up - there's no class hierarchy for the `Builder` objects, since those are only implemented at the concrete resource types, so `toBuilder` and changing objects is more difficult.
I spent a little bit trying to wrangle the types here, but ultimately couldn't come up with anything better than this.